pydantic_settings/sources.py

```diff
@@ -84,9 +84,22 @@ def prepare_field_value(self, field_name: str, field: FieldInfo, value: Any, val
             The prepared value.
         """
         if value is not None and (self.field_is_complex(field) or value_is_complex):
-            return json.loads(value)
+            return self.decode_complex_value(field_name, field, value)
         return value
 
+    def decode_complex_value(self, field_name: str, field: FieldInfo, value: Any) -> Any:
+        """
+        Decode the value for a complex field
+
+        Args:
```

Comment:
Please mention the `field_name` and order them like function args

Reply:
Thanks, resolved :+1: